### PR TITLE
Use char override Writer for System.Private.Xml

### PR DESCRIPTION
**Pull request by kronic**

### CODE REVIEW

**Review by eiriktsarpalis — COMMENTED** [review]
I welcome making the code more consistent --even though any performance benefits are likely theoretical-- but does this actually change all occurrences of single-character strings in the codebase? A simple regex search seems to indicate that there many more that this PR is not touching. Note that this does not limit to `TextWriter.Write` overloads, there also seem to be other internal classes with a similar API pattern, such as `IndentedWriter`.

File: src/libraries/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs [hunk 1]
```diff
@@ -369,7 +369,7 @@ internal static bool GenerateSerializerToStream(XmlMapping[] xmlMappings, Type?[
 
                     writer.Write("[assembly:");
                     writer.Write(typeof(XmlSerializerVersionAttribute).FullName);
-                    writer.Write("(");
+                    writer.Write('(');
                     writer.Write("ParentAssemblyId=");
```
Comment (kasperk81) [inline src/libraries/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs:373]: can join them:
```suggestion
                    writer.Write("(ParentAssemblyId=");
```

Comment (kronic) [reply src/libraries/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs]: @kasperk81 done

**Comment by kronic** [comment]
@eiriktsarpalis done

File: src/libraries/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs (hunk 1 above)
Comment (eiriktsarpalis) [reply src/libraries/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs]: I disagree with this change, it's sacrificing clarity for a minor and theoretical performance improvement. Could you please revert?

Comment (kasperk81) [reply src/libraries/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs]: few lines below there is `WriteLine(")]")`, how is this any different?

Comment (kronic) [reply src/libraries/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs]: @kasperk81 WriteLine  no ovveride char

Comment (kronic) [reply src/libraries/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs]: @eiriktsarpalis I wonder what the performance improvement is. I do not understand?

**Comment by jeffhandley** [comment]
I've marked this PR as "needs more info" for now. We'd like to better understand the risk/reward for the series of XML changes. See https://github.com/dotnet/runtime/pull/61773#issuecomment-973646209. Thanks!

File: src/libraries/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs (hunk 1 above)
Comment (kronic) [reply src/libraries/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs]: @eiriktsarpalis Write(string constant) vs Write(string constant)  + Write(char constant)

File: src/libraries/System.Private.Xml/src/System/Xml/Core/XmlRawWriter.cs [hunk 2]
```diff
@@ -274,7 +274,7 @@ internal virtual void WriteQualifiedName(string prefix, string localName, string
             if (prefix.Length != 0)
             {
                 WriteString(prefix);
-                WriteString(":");
+                WriteCharEntity(':');
```
Comment (drieseng) [inline src/libraries/System.Private.Xml/src/System/Xml/Core/XmlRawWriter.cs:277]: Why would you want to create a character entity here? (and in the other place that you modified)

**Pull request closed without merging**